src/prog_models/sim_result.py

```diff
@@ -97,9 +137,24 @@ def pop(self, index: int = -1) -> dict:
         Returns:
             dict: Element Removed
         """
+        warn('pop will be deprecated after version 1.5 of ProgPy. The function will be renamed, popbyindex, and users may begin using it under this name now.',
+            DeprecationWarning, stacklevel=2)
+
         self.times.pop(index)
         return self.data.pop(index)
 
+    def popbyindex(self, index: int = -1) -> dict:
+        """Remove and return an element
+
+        Args:
+            index (int, optional): Index of element to be removed. Defaults to -1.
+
+        Returns:
+            dict: Element Removed
+        """
+        return self.pop(index)
```

Comment:
This will still raise the deprecation warning when you call pop_by_index. Let's move the logic into this function and have pop call it.